// Copyright 2018 The Chromium Authors. All rights reserved.
// Use of this source code is govered by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

syntax = "proto3";


package monorail;


service Projects {
  rpc ListProjects (ListProjectsRequest) returns (ListProjectsResponse) {}
  rpc UpdateProjectConfiguredLabels (UpdateProjectConfiguredLabelsRequest) returns (Labels) {}
  rpc PatchProjectConfiguredLabels (PatchProjectConfiguredLabelsRequest) returns (Labels) {}
}


message ListProjectsRequest {
  int32 page_size = 1;
  string page_token = 2;
}


message ListProjectsResponse {
  repeated Project projects = 1;
  string next_page_token = 2;
}


message Project {
  string name = 1;
  string summary = 2;
  string description = 3;
}


message UpdateProjectConfiguredLabelsRequest {
  string project = 1;
  Labels labels = 2;
}


message PatchProjectConfiguredLabelsRequest {
  string project = 1;
  repeated Label added_labels = 2;
  repeated Label removed_labels = 3;
}


message Labels {
  repeated Label labels = 1;
}


message Label {
  string name = 1;
  int64 rank = 2;
}
